precice.f90

```diff
@@ -3,8 +3,13 @@ module precice
   implicit none
 
   interface
+
+    ! This interface contains subroutines of two kinds:
+    ! - precicec_<name> expect C-style, null-terminated strings and their sizes
+    ! - precicef_<name> automatically convert strings to null-terminated and call
+    !   the corresponding `precicec_<name>`.
   
-    subroutine precicef_create(participantName, configFileName, &
+    subroutine precicec_create(participantName, configFileName, &
       &                        solverProcessIndex, solverProcessSize, &
       &                        participantNameLength, configFileNameLength) &
       &  bind(c, name='precicef_create_')
```

Comment:
```suggestion
    subroutine precicec_create(participantName, configFileName, &
      &                        solverProcessIndex, solverProcessSize) &
      &  bind(c, name='precicec_createParticipant')
```
If you want to use the C binding and null-terminated strings, it should match the C API:
https://github.com/precice/precice/blob/main/extras/bindings/c/src/preciceC.cpp#L43-L47

Reply:
Ah, I see. So the trick is to target directly the C bindings. Now I get it, thanks!